x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -183,22 +189,42 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 var errTookTooLong = errors.New("could not publish event within timeout")
 
-func getTimeoutWait(u *url.URL, log *logp.Logger) time.Duration {
-	p := u.Query().Get("wait_for_completion_timeout")
+func getTimeoutWait(u *url.URL, log *logp.Logger) (time.Duration, error) {
+	q := u.Query()
+	switch len(q) {
+	case 0:
+		return 0, nil
+	case 1:
+		if _, ok := q["wait_for_completion_timeout"]; !ok {
+			var k string
+			for k = range q {
+				break
+			}
```

Comment:
That confused me a little bit, the `for` loop is there to easily extract the key from the map `q`, is that correct?

Could you add a little comment to make it clear why using a for loop on a map of length 1?

Reply:
I would have thought this was idiomatic, but I will add a comment.